### ISSUE TITLE
[no-magic-numbers] False positives when ignored numbers are in a type definition

### ISSUE THREAD

**Issue by SLKnutson** [typescript-eslint/typescript-eslint#4860]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
"@typescript-eslint/no-magic-numbers": [
      "error",
      { "ignore": [-1, 0, 1, 2], "detectObjects": false, "ignoreReadonlyClassProperties": true, "ignoreEnums": true }
    ]
  }
}
```

```TS
type Something = 1 | 2;
```

**Expected Result**

No type error, 1 and 2 are both ignored

**Actual Result**

`ESLint: No magic number: 1.(@typescript-eslint/no-magic-numbers)`

**Additional Info**

Probably related to https://github.com/typescript-eslint/typescript-eslint/pull/4789/files#diff-4c18ef43d9099bbf45c387ac8b66f16148a091ec5c6abc20e2ff89e44eea5574
Started on 5.19.0 -> 5.20.0 upgrade

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.20.0` |
| `@typescript-eslint/parser`        | `5.20.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.14.0` |
| `node`                             | `16.13.1` |

Thanks!

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-magic-numbers/#ignorenumericliteraltypes

This is working as expected. Use the above option.

**Comment by SLKnutson** [comment]
@bradzacher 
What is the purpose of the "ignore" parameter then?

**Comment by V-ed** [comment]
Also getting this, and even ignoring array indexes doens't seem to do the trick :

```js
// rule
'@typescript-eslint/no-magic-numbers': [
	'warn',
	{ ignoreArrayIndexes: true, ignoreEnums: true, ignoreReadonlyClassProperties: true, ignore: [-1, 0, 1, 1000, 60, 24] },
],
```

![image](https://user-images.githubusercontent.com/16652915/166345074-335281f3-c4d3-4ab9-b3fe-94cc166c631d.png)
![image](https://user-images.githubusercontent.com/16652915/166345131-b69ccbed-5d92-4686-8e30-dbda17f371f8.png)

Meanwhile, the raw eslint rule works :

![image](https://user-images.githubusercontent.com/16652915/166345159-661892f3-7840-471f-928b-2bce67b854dc.png)
![image](https://user-images.githubusercontent.com/16652915/166345181-69ee814c-2e5e-460e-b90d-3a2bf15da9a8.png)

And just to be sure, this is not a eslint inline comment issue, as even using the inline config doesn't work :
![image](https://user-images.githubusercontent.com/16652915/166345290-32645b96-2a72-4812-aa06-ee1c64fc3c19.png)


Sorry for sending many images, I didn't think much on how I would go on about giving a visual example otherwise.

**Comment by V-ed** [comment]
I also confirm this happened in the `5.19.0` => `5.20.0`, as it doesn't happen when using version `5.19.0`.

**Comment by Hypnosphi** [comment]
@V-ed `ignoreTypeIndexes` is what helped me in a similar case. Unfortunately, it still doesn't let you ignore only some specific numbers

**Comment by PhiLhoSoft** [comment]
IMHO, if we choose to leave these options at false, the ignore list should still work, for example for such code:
```
enum CronExpressionMapping {
	Minutes = 1,
	Hours = 2,
	DayOfMonth = 3,
	Month = 4,
	DayOfWeek = 5,
}
```
Before upgrade, it worked as I ignored these values, after upgrade, I had alerts on these lines.

Now, I set `ignoreEnums: true` because it is useful (and logical), but still, I have the feeling the ignore list should be respected otherwise.

**Comment by lo1tuma** [comment]
Unfortunately `ignoreNumericLiteralTypes` only allows to opt out from all magic numbers within literal types. I would like the rule to still flag magic numbers except `0` and `1`.

**Comment by Josh-Cena** [comment]
Closing as a duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/9161 since that one already has a linked PR
